eng/pipelines/templates/stages/archetype-cpp-release.yml

```diff
@@ -164,14 +164,12 @@ stages:
                           CommitMsg: Update vcpkg-configuration.json
                           BaseRepoBranch: $(DefaultBranch)
 
-                      # Set $(HasChanges) to $true so that
-                      # create-pull-request.yml completes the push and PR
-                      # submission steps. This is contegnent upon
-                      # `$(PublishToVcpkg)` being `true`. `$(PublishToVcpkg)` is
-                      # set in `vcpkg-publish.yml`
-                      - pwsh: Write-Host "##vso[task.setvariable variable=HasChanges]$true"
-                        condition: and(succeeded(), eq(variables['PublishToVcpkg'], 'true'))
-                        displayName: Set $(HasChanges) to $true for create-pull-request.yml
+                      # Set $(HasChanges) to $(PublishToVcpkg) so that
+                      # create-pull-request.yml creates or does not create a PR
+                      # based on the deicision of the step that determines
+                      # whether to publish to vcpkg.
+                      - pwsh: Write-Host "##vso[task.setvariable variable=HasChanges]$(PublishToVcpkg)"
+                        displayName: Set $(HasChanges) to $(PublishToVcpkg) for create-pull-request.yml
```

Comment:
@danieljurek Will this render as `Set true to true` or something like that?

Reply:
Macro variable syntax is not expanded in `displayName` . It renders as literal text.

This example shows how it's rendering for `$(HasChanges)` you'll also see a `$true` but that was corrected in this change https://github.com/Azure/azure-sdk-for-cpp/pull/3838/commits/540d806268525da81c4ed4d9aa16536f2bb3edf2 

![image](https://user-images.githubusercontent.com/2158838/180309375-a75f07dd-2fc5-4402-aa20-1c6c1323c1c4.png)

Template syntax does expand into `displayName` For example `Hello ${{ parameters.Location }}` would render as `Hello World` if `parameters.Location = World`

Reply:
Just to avoid that double check look for things you don't expect to expand it might be worth removing the `$()` around them.